File: WMO.WebUi/ClientApp/src/components/Gantt/Task.ts
export interface TaskDto{
    id: number,
    name: string,
    duration: number,
    role: string,
    predecessors: Array<number>
    description?: string
    discipline: string
    phase: string
}


export interface Task{
    id: number,
    name: string,
    duration: number,
    role: string,
    predecessors: Array<number>
    description?: string
    discipline: string
    phase: string
    nexts: Array<Task>,
    prevs: Array<Task>
    offset: number
}

File: WMO.Infrastructure/Models/Task.cs
namespace WMO.Infrastructure.Models
{
    public class Task
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Duration { get; set; }
        public string Role { get; set; }
        public List<int> Predecessors { get; set; }

        public string Description { get; set; }
    }
}

